pkg/depthmonitor/depthmonitor.go

```diff
@@ -128,60 +134,44 @@ func (s *Service) manage(warmupTime time.Duration) {
 		case <-time.After(manageWait):
 		}
 
-		// if we have crossed the adaptation, reset it
-		if time.Now().Sub(adaptationStart).Seconds() > adaptationWindow {
+		currentSize := float64(s.reserve.Size())
+		reserveNotHalfFull := isNotHalfFull(currentSize)
+
+		// if we have crossed 50% utilization, dont do anything
+		if !reserveNotHalfFull {
 			adaptationPeriod = false
+			continue
 		}
 
-		currentSize := float64(s.reserve.Size())
-		capacity := float64(s.reserve.Capacity())
-		reserveNotHalfFull := isNotHalfFull(currentSize, capacity)
-
 		// if we dont have 50% utilization of the reserve, enter into an adaptation period
 		// to see if we need to modify the depth to improve utilization
-		if reserveNotHalfFull && !adaptationPeriod {
+		if !adaptationPeriod {
 			adaptationPeriod = true
 			adaptationStart = time.Now()
 		}
 
-		// if we have crossed 50% utilization, dont do anything
-		if !reserveNotHalfFull {
-			adaptationPeriod = false
+		// if we have crossed the adaptation window, roll it back a little to allow sync to fill the reserve
+		if time.Since(adaptationStart).Seconds() > adaptationWindowSeconds {
+			adaptationStart = time.Now().Add(-time.Second * adaptationRollbackSeconds)
```

Comment:
I dont understand why we need this rollback? 5 seconds are not going to affect the calculations, so I think we should not add this new rollback concept, instead just start a new adaptation period and let the next part use the entire new window to make the calculations.

We can just add a case to the if loop above.